rust/load/src/lib.rs

```diff
@@ -667,20 +857,34 @@ impl LoadService {
             running: Mutex::new(HashMap::default()),
             data_sets,
             workloads,
+            persistent_path: None,
         }
     }
 
+    /// Set the persistent path.
+    pub fn set_persistent_path(&mut self, persistent_path: Option<String>) -> Result<(), Error> {
```

Comment:
I'm a little confused by this method name given it actually loads the persisted state; should `LoadService::new()` take the `persistent_path` as a parameter instead?

Reply:
I didn't make it a parameter because it failing to load should not fail the creation of the struct.  I put it as a fallible method instead so that the error can be logged as a warning and the config dropped.